File: cpp/questions/concatenatedBinary.cpp
class Solution {
private:
    static constexpr int MOD = 1'000'000'007;
public:
    int concatenatedBinary(int n) {
        long long ans = 0;
        for (int cn=1; cn<=n; ++cn) {
            int cnc = std::bit_width(static_cast<uint32_t>(cn));
            ans = ((ans << cnc) | cn) % MOD;
        }
        return ans;
    }
};

File: python/questions/concatenatedBinary.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        MOD = 1000000007
        ans = 0
        for cn in range(1, n+1):
            cnc = cn.bit_length()
            ans = ((ans << cnc) | cn) % MOD
        return ans
